components/eventing-controller/pkg/handlers/jetstream.go

```diff
@@ -101,11 +101,18 @@ func (js *JetStream) Initialize(connCloseHandler ConnClosedHandler) error {
 func (js *JetStream) SyncSubscription(subscription *eventingv1alpha1.Subscription) error {
 	log := utils.LoggerWithSubscription(js.namedLogger(), subscription)
 	subKeyPrefix := createKeyPrefix(subscription)
+	if err := js.checkJetStreamConnection(log); err != nil {
+		return err
+	}
 
 	// check if there is any existing JetStream subscription in global list
 	// which is not anymore in this subscription filters (i.e. cleanSubjects).
 	// e.g. when filters are modified.
 	for key, jsSub := range js.subscriptions {
+		if !jsSub.IsValid() {
```

Comment:
First check if the jssub is related to this kyma subscription and then make changes in the global list of jetstream subs.

Reply:
done